### DISCUSSION TITLE
AttributeError: 'Button' object has no attribute 'message'

### DISCUSSION THREAD

**Discussion by NafezSayyad** [Rapptz/discord.py#9663] [Q&A] [answered]
class MusicControl(discord.ui.View):
    def __init__(self):
        super().__init__()

    @discord.ui.button(label="Pause", style=discord.ButtonStyle.primary, custom_id="pause_button")
    async def pause_button(self, button: discord.ui.Button, interaction: discord.Interaction):
        await interaction.response.send_message(content="Pausing the playback.")
        ctx = await self.bot.get_context(interaction.message)
        await ctx.invoke(self.bot.get_command("pause"))

    @discord.ui.button(label="Skip", style=discord.ButtonStyle.primary, custom_id="skip_button")
    async def skip_button(self, button: discord.ui.Button, interaction: discord.Interaction):
        await interaction.response.send_message("Skipping the current song.")
        ctx = await self.bot.get_context(interaction.message)
        await ctx.invoke(self.bot.get_command("skip"))

    @discord.ui.button(label="Resume", style=discord.ButtonStyle.primary, custom_id="resume_button")
    async def resume_button(self, button: discord.ui.Button, interaction: discord.Interaction):
        await interaction.response.send_message("Resuming the playback.")
        ctx = await self.bot.get_context(interaction.message)
        await ctx.invoke(self.bot.get_command("resume"))

![image](https://github.com/Rapptz/discord.py/assets/62516699/049b7b3f-ce9c-425c-8786-f9dc510ff590)
![image](https://github.com/Rapptz/discord.py/assets/62516699/ca3ffeea-fbd4-4f25-9e22-2461cca96a4c)

**Comment by itslychee** [comment]
Your argument placement for `button` and `interaction` is backwards, the function signature should look like `skip_button(self, interaction: discord.Interaction, button: discord.Button)`

**Comment by Soheab** [comment, marked as answer]
Your parameters are in the wrong order. `interaction` comes before `button`.
```diff
- async def foo(self, button: ui.Button, interaction: Interaction):
+ async def foo(self, interaction: Interaction, button: ui.Button):
```

Consider joining the [discord.py server](https://discord.gg/dpy) and asking for help there in the future but here is fine too.
